### DISCUSSION TITLE
Unclear documentation of query vs. observer options

### DISCUSSION THREAD

**Discussion by d-leeg** [TanStack/query#7523] [General]
The documentation of query options does not clearly state, except in rare occasions (most notably `gcTime` and `staleTime`), which options are at the level of the query (i.e., are associated with a given `queryKey` cache entry), and which are at the level of the observer (i.e., may be different for different observers of the same `queryKey` cache  entry); moreover, for options that are at the query level, the documentation should clearly state how different values specified by different calls to `useQuery` and its friends get reconciled to a single value associated with the query.

For example, while most developers would easily guess that `queryFn` is at the query level, it is unclear what happens if different calls to `useQuery` specify a different `queryFn` for the same key: is the `queryFn` set only by the first call that uses that key (and thus creates the query), and subsequent values ignored, or is it updated every time it is given? It turns the latter is true, but unfortunately undocumented.

The situation is worse with other options where it is much less clear to a developer not familiar with the inner-workings of the library. For example are `retryDelay` or `refetchOnWindowFocus` at the query level or the observer level?

It is quite clear that early on, a decision has been made that the 
API of the library would not make explicit the distinction between the data associated with a given cache entry, (internally called a "query") and an observer of this data (internally called a "queryObserver"). We thus have a single main entry point to create both in the form of the `useQuery` hook. I fully understand this decision, but in order to use the library with confidence the documentation must address this distinction.

**Comment by HasanMothaffar** [comment]
I had the same question as OP. Do `useQuery` options affect observers or cache entries? For example, as for the `enabled` option, if two queries with the same `queryKey` (hence one entry in the cache) have different values for the `enabled` option, what is the expected behavior? The same confusion applies to other options, and I guess it's even worse for `queryFn`

P.S. I'm creating an integration at between firestore subscriptions and react hooks, so I came to this repo for inspiration 👀I learned a lot of new concepts, thanks for the good work.
